client/components/edit/datastream/DatastreamUploadModalContent.tsx

```diff
@@ -1,18 +1,27 @@
 import React from "react";
 import DialogContent from "@mui/material/DialogContent";
+import { useEditorContext } from "../../../context/EditorContext";
+import DatastreamLicenseContent from "./DatastreamLicenseContent";
 import useDatastreamOperation from "../../../hooks/useDatastreamOperation";
 
 const DatastreamUploadModalContent = (): React.ReactElement => {
     const { uploadFile } = useDatastreamOperation();
+    const {
+        state: { activeDatastream },
+    } = useEditorContext();
     const onChange = (event) => {
         uploadFile(event.target.files[0]);
     };
-
-    return (
-        <DialogContent>
-            <input className="uploadFileButton" type="file" onChange={onChange} />
-        </DialogContent>
-    );
+    switch (activeDatastream) {
+        case "LICENSE":
+            return <DatastreamLicenseContent />;
+        default:
+            return (
+                <DialogContent>
```

Comment:
Would it make sense to refactor this to `<DatastreamUploadContent />`? I realize it's short, but that might make sense in the long run (but no pressure if you prefer to leave it as is for now).